### DISCUSSION TITLE
v1.5.0

### DISCUSSION THREAD

**Discussion by milianw** [KDAB/hotspot#622] [Announcements]
# ChangeLog for hotspot v1.5.0

The hotspot v1.5.0 release is finally here :tada:

It comes packed with a wealth of code cleanups, bug fixes and new functionality.
Most notably, the disassembly view has been further improved with better searching,
highlighting and faster performance.

![Screenshot_20240425_220335](https://github.com/KDAB/hotspot/assets/57207/7495adda-a8df-435e-b14d-02071e940d2b)

Furthermore, we reworked the authentication mechanism to allow `perf record` get
run directly with elevated priveleges via `pkexec`, obsoleting the error prone
old mechanism (see also https://nvd.nist.gov/vuln/detail/CVE-2023-28144).

We now also fully support Qt6 and KF6, while keeping compatibility with Qt5
and KF5. The AppImage below is still build with Qt5 but it might be the last
time that we do this. The next version might become Qt6 only.

Many thanks to the various contributors that help build this software, both
by writing code as well as reporting bugs.

Happy profiling everyone :rocket:

## What's Changed
* perfdata added to file open list by @GitMensch in https://github.com/KDAB/hotspot/pull/443
* split README by @GitMensch in https://github.com/KDAB/hotspot/pull/447
* add command line options to README by @GitMensch in https://github.com/KDAB/hotspot/pull/449
* make syntax highlighting customizeable by @lievenhey in https://github.com/KDAB/hotspot/pull/448
* run tests on ubuntu 20.04 ci runner by @lievenhey in https://github.com/KDAB/hotspot/pull/456
* extend appimage debugging notes by @GitMensch in https://github.com/KDAB/hotspot/pull/459
* use non-deprecated function KX11Extras::activeWindow by @lievenhey in https://github.com/KDAB/hotspot/pull/458
* Make libdebugd check in perfparser.cmake take into account library status by @DarkDefender in https://github.com/KDAB/hotspot/pull/465
* Add: QAction to restore default layout by @Da-Viper in https://github.com/KDAB/hotspot/pull/479
* fix sorting order of location in caller callee view by @lievenhey in https://github.com/KDAB/hotspot/pull/468
* improvements for the disassembler by @lievenhey in https://github.com/KDAB/hotspot/pull/457
* allow hotspot to run without gui if --exportTo is supplied by @lievenhey in https://github.com/KDAB/hotspot/pull/481
* Test disassembler highlighting by @lievenhey in https://github.com/KDAB/hotspot/pull/471
* Add link to Fedora repo by @bkmgit in https://github.com/KDAB/hotspot/pull/491
* fix kddockwidgets warning by @lievenhey in https://github.com/KDAB/hotspot/pull/487
* Set source path by @lievenhey in https://github.com/KDAB/hotspot/pull/486
* add search function for source code was: cache source code in vector for easier access in search function by @lievenhey in https://github.com/KDAB/hotspot/pull/470
* New elevate system by @lievenhey in https://github.com/KDAB/hotspot/pull/493
* add option to set custom source paths by @lievenhey in https://github.com/KDAB/hotspot/pull/497
* Set source path fixes by @lievenhey in https://github.com/KDAB/hotspot/pull/499
* remove unused includes by @lievenhey in https://github.com/KDAB/hotspot/pull/504
* remove query stuff from record page by @lievenhey in https://github.com/KDAB/hotspot/pull/454
* use upstream for PrefixTickLabels by @lievenhey in https://github.com/KDAB/hotspot/pull/507
* Search improvements by @lievenhey in https://github.com/KDAB/hotspot/pull/503
* Perf map select by @lievenhey in https://github.com/KDAB/hotspot/pull/510
* add copy function to disassembly views by @lievenhey in https://github.com/KDAB/hotspot/pull/509
* shove flamegraph controls into QToolBar by @lievenhey in https://github.com/KDAB/hotspot/pull/512
* Adjust to KF6 KNotification API change by @kbroulik in https://github.com/KDAB/hotspot/pull/511
* check if perf support libtraceevent by @lievenhey in https://github.com/KDAB/hotspot/pull/520
* add specific version of kf5 to appimage by @lievenhey in https://github.com/KDAB/hotspot/pull/514
* Update perfparser and use configure_file to generate config-perfparser.h by @dfaure-kdab in https://github.com/KDAB/hotspot/pull/517
* fix out of range access by @lievenhey in https://github.com/KDAB/hotspot/pull/530
* Disable branches by @lievenhey in https://github.com/KDAB/hotspot/pull/518
* use python 3.11 to run pre-commit on the ci by @lievenhey in https://github.com/KDAB/hotspot/pull/547
* adjustments to objdump call  by @GitMensch in https://github.com/KDAB/hotspot/pull/535
* Small fixups by @lievenhey in https://github.com/KDAB/hotspot/pull/546
* make clang-tidy happy by @lievenhey in https://github.com/KDAB/hotspot/pull/550
* minor typo correction: unknown by @GitMensch in https://github.com/KDAB/hotspot/pull/540
* run hotspot without GUI for any command-line only option by @GitMensch in https://github.com/KDAB/hotspot/pull/549
* initial gitpod configuration by @GitMensch in https://github.com/KDAB/hotspot/pull/445
* Fix KF6 build by @nicolasfella in https://github.com/KDAB/hotspot/pull/562
* exclude objdump.indexed_start_internal.txt from pre-commit by @lievenhey in https://github.com/KDAB/hotspot/pull/565
* extend old binary filter instead of replacing it by @GitMensch in https://github.com/KDAB/hotspot/pull/559
* don't include non-working context-menu entries by @GitMensch in https://github.com/KDAB/hotspot/pull/558
* limit assembly highlighters by @lievenhey in https://github.com/KDAB/hotspot/pull/567
* add option to hide hexdump by @lievenhey in https://github.com/KDAB/hotspot/pull/566
* peek into file before open by @GitMensch in https://github.com/KDAB/hotspot/pull/539
* add support for kddockwidgets 2.0 by @lievenhey in https://github.com/KDAB/hotspot/pull/570
* use QString overload for KConfigGroup by @lievenhey in https://github.com/KDAB/hotspot/pull/572
* [enhancement] cut 60% time for PerfParser::filterResults by @MuchToMyDelight in https://github.com/KDAB/hotspot/pull/574
* Better asm syntax highlighting by @lievenhey in https://github.com/KDAB/hotspot/pull/538
* add include guard to frequency dock access by @lievenhey in https://github.com/KDAB/hotspot/pull/587
* Disable disassembly of inline functions by @lievenhey in https://github.com/KDAB/hotspot/pull/568
* Enable position independent code for these tests by @amantia in https://github.com/KDAB/hotspot/pull/590
* added test for PERF_FORMAT_LOST attribute by @GitMensch in https://github.com/KDAB/hotspot/pull/579
* CallerCallee add jump to source location in disassembly by @lievenhey in https://github.com/KDAB/hotspot/pull/573
* Fix disassembly navigation by @lievenhey in https://github.com/KDAB/hotspot/pull/593
* Remove the date range from KDAB copyrights by @winterz in https://github.com/KDAB/hotspot/pull/596
* fix jump to disassembly in location view by @lievenhey in https://github.com/KDAB/hotspot/pull/592
* add ci test for non optional build by @lievenhey in https://github.com/KDAB/hotspot/pull/598
* fix wrong source code shown by @lievenhey in https://github.com/KDAB/hotspot/pull/599
* Add error message when exporting a flamegraph that hasn't been opened by @redstrate in https://github.com/KDAB/hotspot/pull/588
* update ci files by @lievenhey in https://github.com/KDAB/hotspot/pull/603
* add vscode applications to the predefined IDEs for "Open in Editor" by @GitMensch in https://github.com/KDAB/hotspot/pull/606
* Fix too many parallel build jobs in appimage build by @Jordan474 in https://github.com/KDAB/hotspot/pull/608
* HighlightedText: Check m_highlighter for nullptr by @kbroulik in https://github.com/KDAB/hotspot/pull/616
* fix typo in disassemblyoutput.h by @lievenhey in https://github.com/KDAB/hotspot/pull/618
* fix multiline highlighting was ~add ci test for non optional build~ by @lievenhey in https://github.com/KDAB/hotspot/pull/594
* fix source code search by @lievenhey in https://github.com/KDAB/hotspot/pull/595

## New Contributors
* @DarkDefender made their first contribution in https://github.com/KDAB/hotspot/pull/465
* @Da-Viper made their first contribution in https://github.com/KDAB/hotspot/pull/479
* @bkmgit made their first contribution in https://github.com/KDAB/hotspot/pull/491
* @kbroulik made their first contribution in https://github.com/KDAB/hotspot/pull/511
* @MuchToMyDelight made their first contribution in https://github.com/KDAB/hotspot/pull/574
* @amantia made their first contribution in https://github.com/KDAB/hotspot/pull/590
* @winterz made their first contribution in https://github.com/KDAB/hotspot/pull/596
* @redstrate made their first contribution in https://github.com/KDAB/hotspot/pull/588
* @Jordan474 made their first contribution in https://github.com/KDAB/hotspot/pull/608

**Full Changelog**: https://github.com/KDAB/hotspot/compare/v1.4.1...v1.5.0

**NOTE:** GitHub does not allow me to upload the AppImage directly anymore, you need to `gunzip` the compressed `AppImage.gz` first :shrug:

<hr /><em>This discussion was created from the release <a href='https://github.com/KDAB/hotspot/releases/tag/v1.5.0'>v1.5.0</a>.</em>

**Comment by GitMensch** [comment]
Can you please add the debuginfo matching the appimage as well?

**Reply by milianw** [reply]
Thanks - done. I forgot that, and since I lost the debuginfo from yesterday I had to rebuild the appimage. You will need to download the new one too if you want to have matching debuginfo - our build isn't reproducible it seems.

**Reply by GitMensch** [reply]
Thanks a lot!

**Comment by GitMensch** [comment]
Do you plan to write an article or a blog entry about this release (like you did for [1.3.0](https://www.kdab.com/hotspot-version-1-3-0-released/) and [1.4.0](https://www.kdab.com/hotspot-v1-4-0/) - or Hoptspot in general, soon?
I'd like to write a LinkedIn post (or an article, but I'm not sure to find the time for that anytime soon) about this software as I find it _extremely_ useful, and would take the opportunity to post some links within that.

And - of course - an updated version of https://www.kdab.com/hotspot-video/ would be marvelous!

**Reply by milianw** [reply]
Yes there will be a blog post, probably released later today on our website.

What part would you like to see updated in the video?

**Reply by GitMensch** [reply]
The whole feature set of `perf annotate` (apart from #528 and _possibly_ #607) got integrated nicely in the meanwhile and at least to me is a real important feature [as well as the callgraph-view (even if it isn't available in the appimage)].
Also the summary is improved and includes the hotspots by binary (which for example allows to quickly exclude "external" binaries from the view [both multi-program and library related]).
